### ISSUE TITLE
select.sf required dplyr > 0.5.0, although I have dplyr 0.7.0

### ISSUE THREAD

**Issue by wgrundlingh** [edzer/sfr#384]
Consider the following example:
```
library(dplyr)
library(sf)
nc <- st_read(system.file('shape/nc.shp', package='sf'))
temp <- nc %>% select(AREA)
```
This throws the error
```
Error in select.sf(., AREA) : 
  requires dplyr > 0.5.0: install that first, then reinstall sf
```
However, `packageVersion('dplyr')` yields `0.7.0`. Any idea what might be going on here?

![image](https://user-images.githubusercontent.com/1776164/27156131-9ee4fc08-5110-11e7-9f64-3b83943039cc.png)

Following the suggestion (to (re)install `dplyr` and then `sf`) doesn't make a difference.

**Comment by edzer** [comment]
I've seen it too, but after installing both pkgs from CRAN I can't reproduce it. Does it help if you start in a clean directory, i.e. without loading an existing .RData file? If not, what is your `devtools::session_info()`?

**Comment by wgrundlingh** [comment]
This is what `devtools::session_info()` returns:

```
> devtools::session_info()
Session info -----------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.0 (2017-04-21)
 system   x86_64, mingw32             
 ui       RStudio (1.0.143)           
 language (EN)                        
 collate  English_Canada.1252         
 tz       America/Los_Angeles         
 date     2017-06-14                  

Packages ---------------------------------------------------------------------------------------------------
 package      * version  date       source        
 assertthat     0.2.0    2017-04-11 CRAN (R 3.3.3)
 base         * 3.4.0    2017-04-21 local         
 cluster        2.0.6    2017-03-10 CRAN (R 3.4.0)
 cobs           1.3-3    2017-03-31 CRAN (R 3.3.3)
 codetools      0.2-15   2016-10-05 CRAN (R 3.4.0)
 colorspace     1.3-2    2016-12-14 CRAN (R 3.3.3)
 compiler       3.4.0    2017-04-21 local         
 datasets     * 3.4.0    2017-04-21 local         
 DBI            0.6-1    2017-04-01 CRAN (R 3.3.3)
 devtools       1.13.2   2017-06-02 CRAN (R 3.4.0)
 digest         0.6.12   2017-01-27 CRAN (R 3.3.3)
 dplyr        * 0.7.0    2017-06-09 CRAN (R 3.4.0)
 fda            2.4.4    2014-12-16 CRAN (R 3.3.3)
 FNN            1.1      2013-07-31 CRAN (R 3.3.3)
 forecast     * 8.0      2017-02-23 CRAN (R 3.3.3)
 fracdiff       1.4-2    2012-12-02 CRAN (R 3.3.3)
 ftsa           4.8      2017-04-25 CRAN (R 3.4.0)
 ggplot2        2.2.1    2016-12-30 CRAN (R 3.3.3)
 glue           1.0.0    2017-04-17 CRAN (R 3.4.0)
 graphics     * 3.4.0    2017-04-21 local         
 grDevices    * 3.4.0    2017-04-21 local         
 grid           3.4.0    2017-04-21 local         
 gtable         0.2.0    2016-02-26 CRAN (R 3.3.3)
 hdrcde         3.1      2013-10-19 CRAN (R 3.3.3)
 htmltools      0.3.6    2017-04-28 CRAN (R 3.3.3)
 htmlwidgets    0.8      2016-11-09 CRAN (R 3.3.3)
 httpuv         1.3.3    2015-08-04 CRAN (R 3.3.3)
 jsonlite       1.5      2017-06-01 CRAN (R 3.3.3)
 KernSmooth     2.23-15  2015-06-29 CRAN (R 3.4.0)
 knitr          1.16     2017-05-18 CRAN (R 3.3.3)
 ks             1.10.6   2017-03-20 CRAN (R 3.3.3)
 lattice        0.20-35  2017-03-25 CRAN (R 3.4.0)
 lazyeval       0.2.0    2016-06-12 CRAN (R 3.3.3)
 lmtest         0.9-35   2017-02-11 CRAN (R 3.3.3)
 magrittr       1.5      2014-11-22 CRAN (R 3.3.3)
 MASS           7.3-47   2017-02-26 CRAN (R 3.4.0)
 Matrix         1.2-9    2017-03-14 CRAN (R 3.4.0)
 MatrixModels   0.4-1    2015-08-22 CRAN (R 3.3.3)
 memoise        1.1.0    2017-04-21 CRAN (R 3.4.0)
 methods      * 3.4.0    2017-04-21 local         
 mgcv           1.8-17   2017-02-08 CRAN (R 3.4.0)
 mime           0.5      2016-07-07 CRAN (R 3.3.2)
 misc3d         0.8-4    2013-01-25 CRAN (R 3.3.3)
 multicool      0.1-10   2016-11-25 CRAN (R 3.3.3)
 munsell        0.4.3    2016-02-13 CRAN (R 3.3.3)
 mvtnorm        1.0-6    2017-03-02 CRAN (R 3.3.2)
 nlme           3.1-131  2017-02-06 CRAN (R 3.4.0)
 nnet           7.3-12   2016-02-02 CRAN (R 3.4.0)
 parallel       3.4.0    2017-04-21 local         
 pcaPP          1.9-61   2016-10-11 CRAN (R 3.3.2)
 plyr           1.8.4    2016-06-08 CRAN (R 3.3.3)
 quadprog       1.5-5    2013-04-17 CRAN (R 3.3.2)
 quantmod       0.4-9    2017-05-29 CRAN (R 3.3.3)
 quantreg       5.33     2017-04-18 CRAN (R 3.3.3)
 R6             2.2.1    2017-05-10 CRAN (R 3.3.3)
 rainbow        3.4      2016-01-03 CRAN (R 3.3.3)
 Rcpp           0.12.11  2017-05-22 CRAN (R 3.3.3)
 rgl            0.98.1   2017-03-08 CRAN (R 3.3.3)
 rlang          0.1.1    2017-05-18 CRAN (R 3.3.3)
 scales         0.4.1    2016-11-09 CRAN (R 3.3.3)
 sde            2.0.15   2016-04-13 CRAN (R 3.3.3)
 sf           * 0.4-3    2017-05-15 CRAN (R 3.4.0)
 shiny          1.0.3    2017-04-26 CRAN (R 3.3.3)
 SparseM        1.77     2017-04-23 CRAN (R 3.3.3)
 splines        3.4.0    2017-04-21 local         
 stats        * 3.4.0    2017-04-21 local         
 stats4         3.4.0    2017-04-21 local         
 tibble         1.3.3    2017-05-28 CRAN (R 3.3.3)
 timeDate       3012.100 2015-01-23 CRAN (R 3.3.2)
 tools          3.4.0    2017-04-21 local         
 tseries        0.10-41  2017-05-29 CRAN (R 3.3.3)
 TTR            0.23-1   2016-03-21 CRAN (R 3.3.3)
 udunits2       0.13     2016-11-17 CRAN (R 3.3.2)
 units          0.4-4    2017-04-20 CRAN (R 3.3.3)
 utils        * 3.4.0    2017-04-21 local         
 withr          1.0.2    2016-06-20 CRAN (R 3.4.0)
 xtable         1.8-2    2016-02-05 CRAN (R 3.3.3)
 xts            0.9-7    2014-01-02 CRAN (R 3.3.3)
 zoo            1.8-0    2017-04-12 CRAN (R 3.3.3)
```

**Comment by edzer** [comment]
As you can see [here](https://github.com/edzer/sfr/commit/f5e38ad35bf0f747e0881759b31bd04431612b16#diff-93435dd5acbfc85b09ff75ef39510db5R133) we defined select.sf doing a version check at build time, rather than at runtime (this, iirc, after advice from @lionel-) but we didn't think it trough enough. It means that for a binary distro (as yours, on win), it it sensitive to the dplyr version available at the time of building sf (which is 0.5-0), rather the one installed on your machine.

To resolve this, you can now install sf from source (either 0.4-x or 0.5-0 from GH), or wait until sf 0.5-0 appears on CRAN (submitted yesterday, but no guarantees).

**Comment by edzer** [comment]
sf 0.5-0 on CRAN now; please reopen if it didn't resolve the issue.
